core/app/assets/javascripts/spree.js

```diff
@@ -22,15 +22,23 @@ Spree.pathFor = function (path) {
 }
 
 Spree.localizedPathFor = function(path) {
-  if (typeof (SPREE_LOCALE) !== 'undefined') {
+  if (typeof (SPREE_LOCALE) !== 'undefined' && SPREE_CURRENCY !== 'undefined') {
+    var currencyParam = '&currency=' + SPREE_CURRENCY
+    var localizedParams = 'locale=' + SPREE_LOCALE + currencyParam
+
     if (path.match(/api\/v/)) {
       if (path.match(/\?/)) {
-        path = path + '&locale=' + SPREE_LOCALE
+        path = path + '&' + localizedParams
       } else {
-        path = path + '?locale=' + SPREE_LOCALE
+        path = path + '?' + localizedParams
       }
     } else {
       path = SPREE_LOCALE + '/' + path
+      if (path.match(/\?/)) {
+        path = path + '&' + currencyParam
+      } else {
+        path = path + '?' + currencyParam
+      }
```

Comment:
Could you please consider refactoring similar blocks of code from line 30 and 37?